lib/src/core_matchers.dart

```diff
@@ -614,25 +615,47 @@ class CustomMatcher extends Matcher {
   featureValueOf(actual) => actual;
 
   bool matches(item, Map matchState) {
-    var f = featureValueOf(item);
-    if (_matcher.matches(f, matchState)) return true;
-    addStateInfo(matchState, {'feature': f});
-    return false;
+    bool rt = false;
+    Chain.capture(() {
+      var f = featureValueOf(item);
+      if (_matcher.matches(f, matchState)) {
+        rt = true;
+        return;
+      }
+      addStateInfo(matchState, {'custom.feature': f});
+    }, onError: (error, stackChain) {
+      addStateInfo(matchState, {
+        'custom.exception': error.toString(),
+        'custom.stack': stackChain.toString()
+      });
+    });
```

Comment:
`Chain.capture()` isn't necessary here, and it won't provide any extra information since it only captures information within the call. You can just use `new Chain.forTrace()` on the stack trace you already captured, and call `.terse.toString()` on it when you print it.